bless/aws_lambda/bless_lambda.py

```diff
@@ -227,6 +209,87 @@ def lambda_handler(event, context=None, ca_private_key_password=None, entropy_ch
+        entropy_check=True,
+        config_file=os.path.join(os.path.dirname(__file__), 'bless_deploy.cfg')):
+    """
+    This is the function that will be called when the lambda function starts.
+    :param event: Dictionary of the json request.
+    :param context: AWS LambdaContext Object
+    http://docs.aws.amazon.com/lambda/latest/dg/python-context-object.html
+    :param ca_private_key_password: For local testing, if the password is provided, skip the KMS
+    decrypt.
+    :param entropy_check: For local testing, if set to false, it will skip checking entropy and
+    won't try to fetch additional random from KMS.
+    :param config_file: The config file to load the SSH CA private key from, and additional settings.
+    :return: the SSH Certificate that can be written to id_rsa-cert.pub or similar file.
+    """
+    bless_cache = setup_lambda_cache(ca_private_key_password, config_file)
+
+    # Load the deployment config values
+    config = bless_cache.config
+
+    logger = set_logger(config)
+
+    ca_private_key = config.getprivatekey()
+
+    # Process cert request
+    schema = BlessHostSchema(strict=True)
+
+    try:
+        request = schema.load(event).data
+    except ValidationError as e:
+        return error_response('InputValidationError', str(e))
+
+    # todo: kmsauth of hostnames? Other server to hostnames validation?
+    logger.info('Bless lambda invoked by [public_key: {}]'.format(request.public_key_to_sign))
+
+    # Make sure we have the ca private key password
+    if bless_cache.ca_private_key_password is None:
+        return error_response('ClientError', bless_cache.ca_private_key_password_error)
+    else:
+        ca_private_key_password = bless_cache.ca_private_key_password
+
+    # if running as a Lambda, we can check the entropy pool and seed it with KMS if desired
+    if entropy_check:
+        check_entropy(config, logger)
+
+    # cert values determined only by lambda and its configs
+    current_time = int(time.time())
+    # todo: config server cert validity range
+    valid_before = current_time + int(timedelta(days=365).total_seconds())  # Host certificate is valid for at least 1 year
+    valid_after = current_time
+
+    # Build the cert
+    ca = get_ssh_certificate_authority(ca_private_key, ca_private_key_password)
+    cert_builder = get_ssh_certificate_builder(ca, SSHCertificateType.HOST,
+                                               request.public_key_to_sign)
+
+    cert_builder.set_valid_before(valid_before)
+    cert_builder.set_valid_after(valid_after)
+
+    # cert_builder is needed to obtain the SSH public key's fingerprint
+    key_id = 'request[{}] ssh_key[{}]  ca[{}] valid_to[{}]'.format(
+        context.aws_request_id, cert_builder.ssh_public_key.fingerprint, context.invoked_function_arn,
+        time.strftime("%Y/%m/%d %H:%M:%S", time.gmtime(valid_before))
+    )
+
+    # todo: verify multiple hostnames are supported in server certs
```

Comment:
you can remove that #todo comment.  multiple hostnames are supported.

Reply:
Yep, tested and working on the client linked.